chall02/gsanchez.py

```diff
@@ -0,0 +1,56 @@
+# **************************************************************************** #
+#                                                                              #
+#                                                         :::      ::::::::    #
+#    gsanchez.py                                        :+:      :+:    :+:    #
+#                                                     +:+ +:+         +:+      #
+#    By: gsanchez <gsanchez@student.42madrid.com    +#+  +:+       +#+         #
+#                                                 +#+#+#+#+#+   +#+            #
+#    Created: 2020/04/06 13:11:23 by gsanchez          #+#    #+#              #
+#    Updated: 2020/04/06 13:56:52 by gsanchez         ###   ########.fr        #
+#                                                                              #
+# **************************************************************************** #
+
+import sys
+
+MORSE_CODE = {"a": ".-",
+        "b": "-...",
+        "c": "-.-.",
+        "d": "-..",
+        "e": ".",
+        "f": "..-.",
+        "g": "--.",
+        "h": "....",
+        "i": "..",
+        "j": ".---",
+        "k": "-.-",
+        "l": ".-..",
+        "m": "--",
+        "n": "-.",
+        "o": "---",
+        "p": ".--.",
+        "q": "--.-",
+        "r": ".-.",
+        "s": "...",
+        "t": "-",
+        "u": "..-",
+        "v": "...-",
+        "w": ".--",
+        "x": "-..-",
+        "y": "-.--",
+        "z": "--..",
+		" ": " ",}
+
+if len(sys.argv) != 2:
+	print("usage " + sys.argv[0] + " <a-zA-Z string>")
+	exit()
+if sys.argv[1] == "":
+	print("usage " + sys.argv[0] + " <a-zA-Z string>")
+	exit()
+for char in sys.argv[1]:
+	if char.lower() not in MORSE_CODE:
+		print("usage " + sys.argv[0] + " <a-zA-Z string>")
+		exit()
+for char in sys.argv[1]:
+	if char.lower() in MORSE_CODE:
+		sys.stdout.write(MORSE_CODE[char.lower()])
+sys.stdout.write('\n')
```

Comment:
Your code is not protected and will be executed even if someone just tries to import part of your code.  
To protect your code and make sure the executable part is only run when we execute the file, you can do this:
```suggestion
if __name__ == "__main__":
    if len(sys.argv) != 2:
    	print("usage " + sys.argv[0] + " <a-zA-Z string>")
    	exit()
    if sys.argv[1] == "":
    	print("usage " + sys.argv[0] + " <a-zA-Z string>")
    	exit()
    for char in sys.argv[1]:
    	if char.lower() not in MORSE_CODE:
    		print("usage " + sys.argv[0] + " <a-zA-Z string>")
    		exit()
    for char in sys.argv[1]:
    	if char.lower() in MORSE_CODE:
    		sys.stdout.write(MORSE_CODE[char.lower()])
    sys.stdout.write('\n')
```
`__name__` is the variable containing the name of the current module, the main module will always be called `__main__`.